Makefile

```diff
@@ -72,6 +126,20 @@ push:
 	docker push $(ORG)/$(IMAGE):$(BRANCH)
 .PHONY: push
 
-generate:
+# Rebuild .proto files
+generate: $(BUF) $(PROTOC)
 	go generate ./...
+	$(BUF) generate
 .PHONY: generate
+
+# Lint .proto files
+lint-protos: $(BUF) $(PROTOC)
+	$(BUF) lint
+.PHONY: lint-protos
+
+# Verify if files built from .proto are up to date.
+check: generate
+	@git add -N $(PROTOC_BUILD_DIRS)
+	@git diff --name-only --diff-filter=AM --exit-code $(PROTOC_BUILD_DIRS) \
+	  || { echo "\nPlease rerun 'make generate' and commit changes.\n"; exit 1; }
```

Comment:
This part now only checks the proto-generation part. It should probably also verify if `go generate ./...` needs to be run again. The point of this check is to make sure that the dev did not forget to regenerate code.

Reply:
It does. It invokes the `generate` target which both calls `go generate` and `$(BUF) generate`

Reply:
But it checks for diff in  $(PROTOC_BUILD_DIRS) only.